src/main/java/com/example/NiRi/service/OrderService.java

```diff
@@ -78,5 +81,20 @@ private void markCartItemsAsPurchased(List<CartItem> cartItems, Order order) {
         SimpleJpaRepository cartItemRepository = null;
         cartItemRepository.saveAll(cartItems);
     }
-}
 
+    public List<Order> getOrdersByUserEmail(String userEmail) {
+        User user = userRepository.findByEmail(userEmail)
+                .orElseThrow(() -> new RuntimeException("User not found with email: " + userEmail));
+
+        return orderRepository.findByUser(user);
+    }
+
+    public Order createOrder(OrderCreationRequest orderRequest) {
+
+        Order newOrder = new Order();
+        newOrder.setUser();
```

Comment:
is this the correct way?

Reply:
I´ve changed it and now the issue is fixed